spec/classes/certs_generate_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+
+describe 'certs::generate' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let :facts do
+        os_facts
+      end
+
+      describe 'with default parameters' do
+        it { should compile.with_all_deps }
+      end
+
+      describe 'with apache true' do
+        let :pre_condition do
+          "class {'certs::generate': apache => true,}"
+        end
+
+        it { should compile.with_all_deps }
```

Comment:
The `should` syntax is deprecated and we should (pun not intended) migrate to `is_expected.to`
```suggestion
        it { is_expected.to compile.with_all_deps }
```